solutions/swb-app/src/accountRoutes.ts

```diff
@@ -17,13 +20,21 @@ import { Request, Response, Router } from 'express';
 import { validate } from 'jsonschema';
 import { escape } from 'lodash';
 import { wrapAsync } from './errorHandlers';
-import { processValidatorResult } from './validatorHelper';
+import { processValidatorResult, validateAndParse } from './validatorHelper';
+
+export function setUpAccountRoutes(router: Router, hostingAccountService: HostingAccountService): void {
+  router.get(
+    '/aws-accounts',
+    wrapAsync(async (req: Request, res: Response) => {
+      const validatedRequest = validateAndParse<ListAccountRequest>(ListAccountsRequestParser, req.query);
```

Comment:
IMO: This sets a bad example for other solutions that wants to use our services because their API would have to have the same field as the service request. Other solutions might want to use different name for the fields in their API. For example, instead of the API request being `{ projects:['proj1']}` the API might be `{ groups: ['group1']}`. This example dictate that the service field must match the API field. This work well for us because we're writing the service and the API.

There should be a `ListAccountsApiRequestParser` that lives in `swb-app/apiModel/account` that describe the shape of the request. In this case it could have the same Zod type definition as  `ListAccountsApiRequestParser` since in our use case the two request are the same. 

We should follow that as an example for future APIs. We shouldn't tightly couple the API request with the service request. 

If we follow this format, DEA can use this as an example for their package `dea-app`. 

An example where they talk about it is [here](https://stackoverflow.com/a/53981773)

I think the code should look something like this
```
// Throw 400 error if request is not valid
validate(ListAccountsApiRequestParser, req.query);
// Map the API Request to the Service Request
const listAccountRequest = getServiceRequest<ListAccountRequest>(ListAccountsRequestParser, req.query, listAccountApiToServiceRequestMapper);

res.send(await hostingAccountService.list(listAccountRequest));
```

With that being said, I'm not going to block the PR on this change. When DEA or another solution wants to use a services in which the service request definition and API definition is different, they can adopt the code example above.